app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventCapturePresenterImpl.java

```diff
@@ -488,6 +492,20 @@ public void nextCalculation(boolean doNextCalculation) {
             Timber.e(calcResult.error());
             return viewModels;
         }
+        List<RuleEffect> testRuleEffect = new ArrayList<>();
+        for(RuleEffect effect : calcResult.items()){
+            RuleAction action = effect.ruleAction();
+            if(action instanceof RuleActionAssign){
+                if(((RuleActionAssign) action).field().startsWith("JGQ")||((RuleActionAssign) action).field().startsWith("gDe")){
+                    testRuleEffect.add(effect);
+                }
+            }else if(action instanceof RuleActionHideField){
+                if(((RuleActionHideField) action).field().startsWith("JGQ")||((RuleActionHideField) action).field().startsWith("gDe")){
+                    testRuleEffect.add(effect);
+                }
+            }
+        }
```

Comment:
It's a problem from the rule engine??? I am worry about the fact that they can have the same problem with another field different than "JGQ" or "gDe" ??